tests/unit_tests/test_registries/test_utils.py

```diff
@@ -1,8 +1,45 @@
 import pytest
-from wandb.apis.public.registries.utils import _ensure_registry_prefix_on_names
+from wandb.apis.public.registries.utils import (
+    _ensure_registry_prefix_on_names,
+    _format_gql_artifact_types_input,
+)
 from wandb.sdk.artifacts._validators import REGISTRY_PREFIX
 
+@pytest.mark.parametrize(
+    "artifact_types, expected_output",
+    [
+        # Valid case
+        (["my-valid-type_123"], [{"name": "my-valid-type_123"}]),
+        # Invalid characters
+        (
+            ["valid_type", "invalid:::"],
+            ValueError,
+        ),
+        (
+            ["invalid/type"],
+            ValueError,
+        ),
+        (
+            ["invalid:type"],
+            ValueError,
+        ),
+        # Too long
+        (["a" * 129], ValueError),
+        # None/empty input
+        (None, []),
+        ([], []),
+    ],
+)
+def test_format_gql_artifact_types_input(artifact_types, expected_output):
+    """Test artifact type name validation and formatting within format_gql_artifact_types_input."""
+    if isinstance(expected_output, type) and issubclass(expected_output, Exception):
+        with pytest.raises(expected_output):
+            _format_gql_artifact_types_input(artifact_types=artifact_types)
+    else:
+        result = _format_gql_artifact_types_input(artifact_types=artifact_types)
+        assert result == expected_output
```

Comment:
feels like it would be more readable to break it into 2 tests instead of this if/else check